src/mame/apple/apple3_v.cpp

```diff
@@ -167,33 +166,25 @@ void apple3_state::text40(bitmap_ind16 &bitmap, const rectangle &cliprect)
 				fg = WHITE;
 			}
 
-			/* inverse? */
-			if (!(ch & 0x80))
-			{
-				using std::swap;
-				swap(fg, bg);
-			}
-
 			uint8_t const *const char_data = &m_char_mem[(ch & 0x7F) * 8];
 
 			for (int row = 0; row < 8; row++)
 			{
 				for (int col = 0; col < 7; col++)
 				{
-					int lc;
-					if (m_smoothscr)
+					uint16_t *const dest = &bitmap.pix(y * 8 + row, x * 14 + col * 2);
+
+					// Flash or Inverse
+					if ((!(ch & 0x80) && (char_data[row] & 0x80) && m_flash) || (!(ch & 0x80) && !(char_data[row] & 0x80)))
 					{
```

Comment:
This seems to have excess parentheses and common subexpressions.  Wouldn’t it be clearer as:
```c++
 		if (!(ch & 0x80) && (!(char_data[row] & 0x80) || m_flash))
```

That’s equivalent logic after minimising the truth table.

Reply:
yes, that's much better.

Reply:
updated